amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -119,6 +124,7 @@ public TaskResult call() {
                     if (!CollectionUtils.isNullOrEmpty(childShards)) {
```

Comment:
As called out below, we should differentiate between ResourceNotFound case and the case of shard shutdown eligible for cleanup.

Reply:
Added state info as per earlier comment.